kit/src/main/java/com/oracle/javafx/scenebuilder/kit/editor/DocumentationUrls.java

```diff
@@ -0,0 +1,247 @@
+ */
+package com.oracle.javafx.scenebuilder.kit.editor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Documentation URLs are provided in a resource
+ * {@code documentation_urls.properties}. These URLs point to OpenJFX pages such
+ * as Javadoc API / CSS or FXML references. Those are meant to be used in the
+ * Scene Builder help menu to eventually provide quick and easy access to
+ * essential JavaFX developer documentation.
+ */
+public class DocumentationUrls {
+
+    private static final String UNKNOWN = "unknown";
+
+    private static DocumentationUrls instance = null;
+
+    public static DocumentationUrls getInstance() {
+        if (instance == null) {
+            instance = new DocumentationUrls();
+        }
+        return instance;
+    }
+
+    /**
+     * Verifies if the desired {@link DocumentationItem} is available.
+     * 
+     * @param item {@link DocumentationItem} (usually an URL)
+     * @return true when available in corresponding resource
+     */
+    public static boolean isAvailable(DocumentationItem item) {
+        return getInstance().getOptionalUrl(item).isPresent();
+    }
+
+    /**
+     * Detects the used JavaFX version from system properties using
+     * {@code System.getProperty("javafx.version")}. In case this property is
+     * undefined, the String "unknown" is provided.
+     * 
+     * @return JavaFX version number or "unknown" when property is undefined
+     */
+    protected static String getJavaFxVersion() {
+        return getJavaFxVersion(System.getProperties());
+    }
+
+    /**
+     * Detects the used JavaFX version from given properties instance using property
+     * key {@code "javafx.version"}. In case this property is undefined, the String
+     * "unknown" is provided.
+     * 
+     * @return JavaFX version number or "unknown" when property is undefined
+     */
+    protected static String getJavaFxVersion(Properties systemProperties) {
+        Objects.requireNonNull(systemProperties);
+        String fxVersion = systemProperties.getProperty("javafx.version");
+        if (fxVersion == null) {
+            return UNKNOWN;
+        }
+        return fxVersion;
+    }
+
+    /**
+     * @param version JavaFX version string
+     * @return Provides the major version of the given version String.
+     */
+    protected static String getMajorJavaFxVersion(String version) {
+        if (version == null || version.isBlank()) {
+            return UNKNOWN;
+        }
+        int firstDot = version.indexOf('.');
+        if (firstDot > 1) {
+            return version.substring(0, firstDot);
+        } else {
+            return version;
+        }
+    }
+
+    private final Map<DocumentationItem, String> configuredUrls;
+
+    private final String javaFxVersion;
+
+    DocumentationUrls() {
```

Comment:
Why isn't the constructor declared as `private`?

Reply:
This one can sure be private. So I'll put this private.

Reply:
Well the reason this is package visible is for the test, so that one could test the default behavior without relying on the singleton. Well, I could remove the test and reduce the visibilty accordingly.

Reply:
The test now uses `getInstance`, hence the constructor is now private.